geomesa-redis/geomesa-redis-datastore/src/main/scala/org/locationtech/geomesa/redis/data/RedisDataStoreFactory.scala

```diff
@@ -90,31 +91,68 @@ object RedisDataStoreFactory extends GeoMesaDataStoreInfo with LazyLogging {
     * @param params params
     * @return
     */
-  def buildConnection(params: java.util.Map[String, _]): JedisPool = {
+  def buildConnection(params: java.util.Map[String, _]): Pool[_ <: CloseableJedisCommands] = {
     ConnectionPoolParam.lookupOpt(params).getOrElse {
-      val uri = {
-        val url = RedisUrlParam.lookup(params)
-        // if there is no protocol/port, or the url is a valid redis url, use as is
-        // else use the redis:// protocol to support databases, etc
-        val parsed =
-          if (url.indexOf(":") == -1) {
-            Try(new URI(url))
-          } else {
-            parse(url).orElse(parse(s"redis://$url"))
+      RedisClusterBoolParam.lookup(params).booleanValue() match {
+        case false =>
+          val uri = {
+            val urls = RedisUrlParam.lookup(params).split(",")
+            if (urls.length > 1) {
+              throw new IllegalArgumentException(
+                "Multiple Redis URLs can only be used in cluster mode. Set 'redis.clusterMode' to true.")
+            }
+            val url = urls.headOption.getOrElse {
+              throw new IllegalArgumentException("No Redis URL provided. Please set 'redis.url'")
+            }
+            // if there is no protocol/port, or the url is a valid redis url, use as is
+            // else use the redis:// protocol to support databases, etc
+            val parsed =
+              if (url.indexOf(":") == -1) {
+                Try(new URI(url))
+              } else {
+                parse(url).orElse(parse(s"redis://$url"))
+              }
+            parsed match {
+              case Success(uri) => uri
+              case Failure(e) =>
+                throw new IllegalArgumentException(s"Could not create valid Redis connection URI from: $url", e)
+            }
           }
-        parsed match {
-          case Success(uri) => uri
-          case Failure(e) =>
-            throw new IllegalArgumentException(s"Could not create valid Redis connection URI from: $url", e)
-        }
+          val config = new GenericObjectPoolConfig[Jedis]()
+          PoolSizeParam.lookupOpt(params).foreach(s => config.setMaxTotal(s.intValue()))
+          config.setTestOnBorrow(TestConnectionParam.lookup(params))
+          val timeout = SocketTimeoutParam.lookup(params).toMillis.toInt
+
+          new JedisPool(config, uri, timeout)
+        case true =>
+          val urls = RedisUrlParam.lookup(params).split(",")
+          val uri = URI.create(urls.head)
+          if (urls.isEmpty) {
+            throw new IllegalArgumentException("No Redis URLs provided. Please set 'redis.url'")
+          }
```

Comment:
i think these 2 lines need to be flipped:

```suggestion
          if (urls.isEmpty) {
            throw new IllegalArgumentException("No Redis URLs provided. Please set 'redis.url'")
          }
          val uri = URI.create(urls.head)
```